numpy/random/mtrand.pyx

```diff
@@ -4462,6 +4462,17 @@ cdef class RandomState:
                 # shuffling is a no-op
                 return
 
+            if x.ndim == 1 and x.dtype.type is np.object_:
+                warnings.warn(
+                        "Shuffling a one dimensional array subclass containing "
+                        "objects gives incorrect results for most array "
```

Comment:
```suggestion
                        "Shuffling a one dimensional array subclass of dtype "
                        "object gives incorrect results for most array "
```
since structured types containing objects are fine.